api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -588,6 +588,9 @@ message Cluster {
       // requests as if all hosts are unhealthy. This can help avoid potentially overwhelming a
       // failing service.
       bool fail_traffic_on_panic = 3;
+
+      // If set to true, Envoy will force LocalityDirect routing if a local locality exists.
```

Comment:
will this bypass min_cluster_size or respect it ?

Reply:
Still WIP but am planning to add a companion option that configures the minimum number of hosts in the local zone.